packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ClientModelPlugin.cs

```diff
@@ -25,6 +25,12 @@ public class ClientModelPlugin : CodeModelPlugin
 
         public override ExtensibleSnippets ExtensibleSnippets { get; }
 
+        /// <summary>
+        /// Returns a serialization type provider of type <see cref="SystemModelSerializationTypeProvider"/> for the given model type provider.
+        /// </summary>
+        /// <param name="provider">The model type provider.</param>
+        public override TypeProvider? GetSerializationTypeProvider(ModelTypeProvider provider) => new SystemModelSerializationTypeProvider(provider);
```

Comment:
So I think if the serialization has to get from the model, why we do not put this on the ModelTypeProvider because the serialization belongs to its model

Reply:
ModelTypeProvider is currently sealed.

Reply:
It needs to be something that customers can replace with their own implementation.

Reply:
> ModelTypeProvider is currently sealed.

anyhow - i ask this because I put this inside `EnumTypeProvider` for enums, and `EnumTypeProvider` is not sealed.

So if we would like plugin writers to extend/change the implementation, why `ModelTypeProvider` be sealed?
@m-nash so one of the scenarios you mentioned that someone would like to write a plugin is that they might do not want the collection properties to be "not settable", I do not think they could do it intuitively if `ModelTypeProvider` is sealed.

But whatever you decided here, I would align in my implementation of enums. Please let me know what it ends up to be